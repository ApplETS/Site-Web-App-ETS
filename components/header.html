<nav>
  <a href="#main" class="skip-link">Aller au contenu principal</a>

  <div class="top-row">
    <a href="index.html"><img src="images/applets.svg" alt=""> </a>
    <button type="button" id="nav-toggle-container" class="btn-transparent" aria-label="Menu principal" aria-expanded="false" aria-controls="nav-list">
      <div id="nav-toggler">
        <span></span>
      </div>
    </button>
  </div>

	<ul id="nav-list" style="height: 0; display: none;">
		<li><a href="index.html">Accueil</a></li>
		<li><a href="equipe.html">Ã‰quipe</a></li>
    <li><a href="partenaires.html">Partenaires</a></li>
    <li><a href="contact.html">Contact</a></li>
    <li><a href="https://formulaires.etsmtl.ca/ClubEtudiantAdhesion" target="_blank" id="btn-membre">Devenir Membre</a></li>
	</ul>
</nav> 