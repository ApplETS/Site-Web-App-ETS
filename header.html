<div class="top-row">
	<img src="images/applets.svg" alt="">

	<button id="nav-toggle-container" class="btn-transparent" aria-label="Menu">
	  <div id="nav-toggler">
		<span></span>
	  </div>
	</button>
</div>

<nav id="nav-toggle-content" style="display: none;">
	<ul>
		<li><a href="index.html">Accueil</a></li>
		<li><a href="equipe.html">Ã‰quipe</a></li>
		<li><a href="partenaires.html">Partenaires</a></li>
		<li><a href="contact.html">Contact</a></li>
	</ul>
</nav>
<script src="js/navbar.js"></script>