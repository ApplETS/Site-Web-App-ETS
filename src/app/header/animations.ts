import { animate, animation, query, style } from '@angular/animations';

export const hideButtons = animation([
  style({ backgroundColor: 'transparent' }),
  query('.nav-item', [
    style({ opacity: 0 })
  ])
]);

export const fadeIn = animation([
  query('#navTitle', [
    style({ opacity: 0 }),
    animate('{{ time }}'),
    style({ opacity: 1 })
  ])
]);
