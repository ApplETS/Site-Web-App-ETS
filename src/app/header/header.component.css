.navbar {
  width: 100%;
  background-color: #1c1c1d;
  justify-content: center;
  align-items: center;
  color: white;
  font-size: 2vh;
  display: flex;
  padding: 0.5rem 1rem;
}

.navbar-brand {
  padding-top: 0.3125rem;
  padding-bottom: 0.3125rem;
  margin-right: 1rem;
}

.nav-buttons {
  padding: 0;
}

.nav-buttons li:hover::after {
  width: 100%;
}

.nav-buttons li.currentTab:after {
  width: 100%;
}

.nav-buttons li {
  cursor: pointer;
  margin-left: 7vh;
  display: inline-block;
}

.nav-buttons li:after {
  content: '';
  display: block;
  width: 0;
  height: 2px;
  background: #fff;
  transition: width 0.3s;
}

.nav-buttons li:hover::after {
  width: 100%;
}

.nav-buttons li.active:after {
  width: 100%;
}

.nav-item a {
  text-decoration: none;
  color: inherit;
}

.brandImage {
  height: 7vh;
}
