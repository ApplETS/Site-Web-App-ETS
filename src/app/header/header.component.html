<header class="navbar" [@fade]="'start'">
  <span id="navTitle" class="navbar-brand" style="font-size:5vh;cursor:default;color:white;">
    <img id="brandImage" class="brandImage" src="assets/images/Logo_blanc_ApplETS.png" alt="Logo du club App|ETS" />
  </span>
  <div id="floatingToggle"></div>
  <a id="nav-toggler">
    <span></span>
  </a>

  <div class="main-nav" id="navToggleContent">
    <ul id="navButtons" class="nav-buttons navbar-nav" [@buttons]="'start'">
      <li class="nav-item">
        <a routerLink="/">Accueil</a>
      </li>
      <li class="nav-item">
        <a href="equipe.html">Ã‰quipe</a>
      </li>
      <li class="nav-item">
        <a href="projets.html">Projets</a>
      </li>

      <li class="nav-item">
        <a href="partenaires.html">Partenaires</a>
      </li>
      <li class="nav-item">
        <a href="contact.html">Contact</a>
      </li>
    </ul>
  </div>
</header>
