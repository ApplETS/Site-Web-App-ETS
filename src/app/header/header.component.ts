import { Component, OnInit } from '@angular/core';
import { animate, query, stagger, style, transition, trigger, useAnimation } from '@angular/animations';
import { fadeIn, hideButtons } from './animations';

const swingAnimation = trigger('fade', [
  transition(':enter', [
    useAnimation(hideButtons),
    useAnimation(fadeIn, {
      params: {
        time: '1s'
      }
    }),
    query('.nav-item', [stagger(100, [style({ opacity: 0 }), animate(200), style({ opacity: 1 })])]),
    animate(500),
    style({ opacity: 1 })
  ])
]);

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.css'],
  animations: [swingAnimation]
})

/**
 * This component represent the header of the web site with the navigation menu
 */
export class HeaderComponent implements OnInit {
  constructor() {}

  ngOnInit() {}
}
