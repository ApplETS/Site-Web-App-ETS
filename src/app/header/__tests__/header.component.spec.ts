import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { HeaderComponent } from '../header.component';

describe('HeaderComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [HeaderComponent],
      imports: [BrowserAnimationsModule]
    }).compileComponents();

    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should matched the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
