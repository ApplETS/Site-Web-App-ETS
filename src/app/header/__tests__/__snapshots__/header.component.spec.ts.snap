// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should matched the snapshot 1`] = `
<app-header>
  <header
    class="navbar ng-trigger ng-trigger-fade"
    style=""
  >
    <span
      class="navbar-brand"
      id="navTitle"
      style="font-size:5vh;cursor:default;color:white;"
    >
      <img
        alt="Logo du club App|ETS"
        class="brandImage"
        id="brandImage"
        src="assets/images/Logo_blanc_ApplETS.png"
      />
    </span>
    <div
      class="ng-tns-c1-0"
      id="floatingToggle"
    />
    <div
      class="main-nav"
      id="navToggleContent"
    >
      <ul
        class="nav-buttons navbar-nav"
        id="navButtons"
      >
        <li
          class="nav-item"
          routerlinkactive="active"
        >
          <a
            class="ng-tns-c1-0"
            routerlink="/home"
          >
            Accueil
          </a>
        </li>
        <li
          class="nav-item"
          routerlinkactive="active"
        >
          <a
            class="ng-tns-c1-0"
            href="equipe.html"
          >
            Ã‰quipe
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="ng-tns-c1-0"
            href="projets.html"
          >
            Projets
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="ng-tns-c1-0"
            href="partenaires.html"
          >
            Partenaires
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="ng-tns-c1-0"
            href="contact.html"
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
  </header>
</app-header>
`;
